### PR TITLE
[test helper] add function to enable initializing and minting of aptos coin with genesis

### PR DESCRIPTION
**Pull request by mutable-dev**

### CODE REVIEW

File: aptos-move/framework/aptos-framework/sources/aptos_coin.move [hunk 1]
```diff
@@ -159,4 +159,9 @@ module aptos_framework::aptos_coin {
         aggregator_factory::initialize_aggregator_factory_for_test(aptos_framework);
         initialize(aptos_framework)
     }
+
+    #[test_only]
+    public fun initialize_for_test_without_aggregator_factory(aptos_framework: &signer): (BurnCapability<AptosCoin>, MintCapability<AptosCoin>) {
+        initialize(aptos_framework)
+    }
```
Comment (gregnazario) [inline aptos-move/framework/aptos-framework/sources/aptos_coin.move:166]: I'm just curious, does how does the one with the aggregator factory not support your testing case?

Comment (movekevin) [reply aptos-move/framework/aptos-framework/sources/aptos_coin.move:166]: +1

File: aptos-move/framework/aptos-framework/sources/aptos_coin.move [hunk 2]
```diff
@@ -159,4 +159,9 @@ module aptos_framework::aptos_coin {
         aggregator_factory::initialize_aggregator_factory_for_test(aptos_framework);
         initialize(aptos_framework)
     }
+
+    #[test_only]
```
Comment (davidiw) [inline aptos-move/framework/aptos-framework/sources/aptos_coin.move:163]:
```suggestion
    // This is particularly useful if the aggregator_factory is already initialized via another call path.
    #[test_only]
```

**Review by davidiw — APPROVED** [review]

File: aptos-move/framework/aptos-framework/sources/aptos_coin.move (hunk 1 above)
Comment (davidiw) [reply aptos-move/framework/aptos-framework/sources/aptos_coin.move:167]: one cannot initialize genesis and get access to aptos_coin ... we had an oversight.

**Review by gregnazario — APPROVED** [review]

**Pull request merged**